src/app/workflow/view/view.component.ts

```diff
@@ -100,6 +100,9 @@ export class ViewWorkflowComponent extends View implements OnInit {
    * @memberof ViewWorkflowComponent
    */
   private updateWorkflowToSnapshot(version): void {
+    // Clear sourcefiles to shrink version
+    version.sourceFiles = [];
```

Comment:
I think you should create a new object; if the operation fails for whatever reason, sourceFiles would be empty and would for example affect the Files tab. Unless this already is a copy... Not sure.

Reply:
According to Gary, it is a clone. I will do some more testing locally before I merge this.

Reply:
So this is the value that is passed to the function
`const snapshot: WorkflowVersion = { ...this.version, frozen: true };`

Here is the explanation of the spread operator in this case https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals

Here is the relevant line about the spread

> It copies own enumerable properties from a provided object onto a new object